platforms/software/dependency-management/src/main/java/org/gradle/api/internal/notations/DependencyClassPathNotationConverter.java

```diff
@@ -83,16 +80,15 @@ public void convert(DependencyFactoryInternal.ClassPathNotation notation, Notati
     }
 
     private FileCollectionDependency create(final DependencyFactoryInternal.ClassPathNotation notation) {
-        boolean runningFromInstallation = currentGradleInstallation.getInstallation() != null;
         FileCollectionInternal fileCollectionInternal;
-        if (runningFromInstallation && notation.equals(GRADLE_API)) {
+        if (notation.equals(GRADLE_API)) {
```

Comment:
What changed here?

Reply:
After changing the `module-jar` plugin to use artifactView instead of the deprecated fileCollection filtering method, this test started failing on embedded only:  

https://github.com/gradle/gradle/blob/22ebab694b7f880d62bf64941ed3f15816670fe3/platforms/extensibility/test-kit/src/integTest/groovy/org/gradle/testkit/TestKitDependencyClassVisibilityIntegrationTest.groovy#L41-L56

That led me to this code, where where I found that when running in embedded, we _don't_ use the shaded jar for testkit dependencies. This change makes the testkit dependency behave the same between embedded and forking tests (same with gradle API dependency)